File: app/src/main/java/com/adiaz/deportesmadrid/adapters/expandable/WeekGroup.java
package com.adiaz.deportesmadrid.adapters.expandable;

import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup;

import java.util.List;

public class WeekGroup extends ExpandableGroup<MatchChild> {

    private Integer idWeek;

    public WeekGroup(String title, List<MatchChild> items, Integer idWeek) {
        super(title, items);
        this.idWeek = idWeek;
    }

    public Integer getIdWeek() {
        return idWeek;
    }

    public void setIdWeek(Integer idWeek) {
        this.idWeek = idWeek;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof WeekGroup)) return false;
        WeekGroup genre = (WeekGroup) o;
        return getTitle() == genre.getTitle();
    }

    @Override
    public int hashCode() {
        return getTitle().hashCode();
    }

}

File: app/src/main/java/com/adiaz/ligasmadrid/adapters/expandable/WeekGroup.kt
package com.adiaz.ligasmadrid.adapters.expandable

import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup

class WeekGroup(title: String, items: MutableList<MatchChildKotlin>, var idWeek: Int?) : ExpandableGroup<MatchChildKotlin>(title, items) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is WeekGroup) return false
        val genre = o as WeekGroup?
        return title === genre!!.title
    }

    override fun hashCode(): Int {
        return title.hashCode()
    }

}
